st2common/st2common/runners/paramiko_ssh.py

```diff
@@ -579,13 +579,42 @@ def _connect(self, host, socket=None):
         if socket:
             conninfo['sock'] = socket
 
+        if cfg.CONF.ssh_runner.use_ssh_config:
+            conninfo_host_ssh_config = self._get_conninfo_from_ssh_config_for_host(host)
+            conninfo.update(conninfo_host_ssh_config)
```

Comment:
That is correct @kami. If we have `use_ssh_config = True`, values from config have precedence over action's parameters. There is one more caveat: `core.remote` fails with `hosts=localhost`.